cmd/finch/virtual_machine_status_test.go

```diff
@@ -0,0 +1,126 @@
+		wantErr error
+		groups  func(*gomock.Controller) []*dependency.Group
+		mockSvc func(
+			*mocks.LimaCmdCreator,
+			*mocks.Logger,
+			*mocks.LimaConfigApplier,
+			*gomock.Controller,
+		)
+	}{
+		{
+			name:    "running VM",
+			wantErr: nil,
+			groups: func(ctrl *gomock.Controller) []*dependency.Group {
+				return nil
+			},
+			mockSvc: func(
+				lcc *mocks.LimaCmdCreator,
+				logger *mocks.Logger,
+				lca *mocks.LimaConfigApplier,
+				ctrl *gomock.Controller,
+			) {
+				getVMStatusC := mocks.NewCommand(ctrl)
+				lcc.EXPECT().CreateWithoutStdio("ls", "-f", "{{.Status}}", limaInstanceName).Return(getVMStatusC)
+				getVMStatusC.EXPECT().Output().Return([]byte("Running"), nil)
+				logger.EXPECT().Debugf("Status of virtual machine: %s", "Running")
+			},
+		},
+		{
+			name: "stopped VM",
+			wantErr: nil,
+			groups: func(ctrl *gomock.Controller) []*dependency.Group {
+				return nil
+			},
+			mockSvc: func(
+				lcc *mocks.LimaCmdCreator,
+				logger *mocks.Logger,
+				lca *mocks.LimaConfigApplier,
+				ctrl *gomock.Controller,
+			) {
+				getVMStatusC := mocks.NewCommand(ctrl)
+				lcc.EXPECT().CreateWithoutStdio("ls", "-f", "{{.Status}}", limaInstanceName).Return(getVMStatusC)
+				getVMStatusC.EXPECT().Output().Return([]byte("Stopped"), nil)
+				logger.EXPECT().Debugf("Status of virtual machine: %s", "Stopped")
+			},
+		},
+		{
+			name:    "unknown VM status",
+			wantErr: errors.New("unrecognized system status"),
+			groups: func(ctrl *gomock.Controller) []*dependency.Group {
+				return nil
+			},
+			mockSvc: func(
+				lcc *mocks.LimaCmdCreator,
+				logger *mocks.Logger,
+				lca *mocks.LimaConfigApplier,
+				ctrl *gomock.Controller,
+			) {
+				getVMStatusC := mocks.NewCommand(ctrl)
+				lcc.EXPECT().CreateWithoutStdio("ls", "-f", "{{.Status}}", limaInstanceName).Return(getVMStatusC)
+				getVMStatusC.EXPECT().Output().Return([]byte("Broken"), nil)
+				logger.EXPECT().Debugf("Status of virtual machine: %s", "Broken")
+			},
+		},
+		{
+			name:    "status command returns an error",
+			wantErr: errors.New("get status error"),
+			groups: func(ctrl *gomock.Controller) []*dependency.Group {
+				return nil
+			},
+			mockSvc: func(
+				lcc *mocks.LimaCmdCreator,
+				logger *mocks.Logger,
+				lca *mocks.LimaConfigApplier,
+				ctrl *gomock.Controller,
+			) {
+				getVMStatusC := mocks.NewCommand(ctrl)
+				lcc.EXPECT().CreateWithoutStdio("ls", "-f", "{{.Status}}", limaInstanceName).Return(getVMStatusC)
+				getVMStatusC.EXPECT().Output().Return([]byte("Broken"), errors.New("get status error"))
+			},
+		},
+	}
+
+	for _, tc := range testCases {
+		tc := tc
+		t.Run(tc.name, func(t *testing.T) {
+			t.Parallel()
+
+			ctrl := gomock.NewController(t)
+			logger := mocks.NewLogger(ctrl)
+			lcc := mocks.NewLimaCmdCreator(ctrl)
+			lca := mocks.NewLimaConfigApplier(ctrl)
+
+			tc.mockSvc(lcc, logger, lca, ctrl)
+
+			err := newStatusVMAction(lcc, logger).run()
+			assert.Equal(t, err, tc.wantErr)
+		})
```

Comment:
We also need to test the stdout here. It is not straightforward but this is the [example](https://github.com/runfinch/finch/blob/main/cmd/finch/version_test.go#L47).

Maybe we can extract the common method captureStdout to somewhere like a new package `fio`. @davidhsingyuchen may give good suggestions about where to extract to.

Reply:
I tried implementing it like:
```go
statusActionFunc := func() {
	err := newStatusVMAction(lcc, logger).run()
	assert.Equal(t, err, tc.wantErr)
}

output := captureStdout(t, statusActionFunc)
assert.Equal(t, tc.wantStatusOutput, output)
```
For the sake of simplicity and testing I used the method already implemented. Let me know how we want to implement the `fio` package so I can use it in the right way. 

My problem is now that the tests are really flaky. Sometimes they pass sometimes they don't. 
One try it is:
```
expected: "Running\n"
actual  : ""
```
Next one it passes. 

Does this happen because of the parallel tests?

Reply:
We can and probably should replace `captureStdout` with dependency injection. We can pass an `io.Writer` to `newStatusVMAction`, which will be `os.Stdout` in the real code and a `*bytes.Buffer` in the testing code. Demonstrating the relevant part with some real code:

```go
func test() {
	var b bytes.Buffer
	status(&b) // would be status(os.Stdout) in the real implementation (i.e., when not testing)
	if string(b.Bytes()) == "running\n" { // assert whatever we want here, please use the assert lib in the real testing code
		fmt.Println("pass")
	} else {
		fmt.Println("fail")
	}
}

func status(stdout io.Writer) {
	fmt.Fprintln(stdout, "running")
}
```

Reply:
It is not common to have flaky unit tests. Could you try the dependency injection approach? If it is still flaky, you can push the commit to the PR so we can take a look. @niklasmtj